### PR TITLE
Add MD property interface and granular capabilities

### PR DESCRIPTION
**Pull request by snschune**

### CODE REVIEW

**Comment by snschune** [comment]
This will need some additional work so no review at this point.

**Comment by snschune** [comment]
1. @dschwen what do you think is the best way to include overlap in this PR?
2. Add a `particle_radius` command to enable initial partitioning with granular particles. 
3. This should read actual LAMMPS files as well and not just dump files.

I'm gonna do 1 & 2 before this MR can go in.

**Comment by snschune** [comment]
@dschwen before this can be merged, do we need to acknowledge the new contrib, `overlap.hpp`, somehow?

**Comment by dschwen** [comment]
Lets add 

```
Part of this code uses the exact calculation of the overlap volume of spheres and mesh elements routines copyright (C) 2015-2018 by Severin Strobl <git@severin-strobl.de> (http://dx.doi.org/10.1016/j.jcp.2016.02.003)
```

to the `README.md` file

**Comment by dschwen** [comment]
maybe with a markdown link to `contrib/overlap/README.md`

File: src/userobjects/MDRunBase.C [hunk 1]
```diff
@@ -39,35 +39,78 @@ validParams<MDRunBase>()
...
+  {
+    if (!parameters.isParamSetByUser("max_granular_radius"))
+      paramError("max_granular_radius",
+                 "max_granular_radius must be set for granular calculations");
+    _max_granular_radius = getParam<Real>("max_granular_radius");
+  }
+
+  // check _properties array and MooseEnum for consistency
+  if (N_MD_PROPERTIES != mdParticleProperties().getNames().size())
+    mooseError("Mismatch of MD particle property enum and property array. Report on github.");
```
Comment (dschwen) [inline src/userobjects/MDRunBase.C]: Oh, does that mean a very specific dump format is required?

Comment (snschune) [reply src/userobjects/MDRunBase.C]: Yes, must be `xyz` format.

**Comment by snschune** [comment]
All comments should be addressed @dschwen.

**Pull request merged**